src/test/scala/com/cognite/sdk/scala/common/DataModelPropertiesSerializerTest.scala

```diff
@@ -0,0 +1,536 @@
+              Vector(1.2, 2, 4.654)
+            ),
+            "arr_int32" -> PropertyType.Array.Int.Property(
+              Vector(3, 1, 2147483646)
+            ),
+            "arr_int64" -> PropertyType.Array.Bigint.Property(
+              Vector(2147483650L, 0, 9223372036854775L, 1).map(BigInt(_))
+            ),
+            "arr_string" -> PropertyType.Array.Text.Property(
+              Vector("tata", "titi")
+            ),
+            "arr_empty" -> PropertyType.Array.Text.Property(Vector()),
+            "arr_empty_nullable" -> PropertyType.Array.Float64.Property(Vector())
+          )
+      }
+      "work for nullable property" in {
+        val res = decode[PropertyMap]("""{
+                                        |    "externalId": "test",
+                                        |    "prop_float64": 23.0,
+                                        |    "prop_string": "toto",
+                                        |    "prop_direct_relation": "Asset",
+                                        |    "arr_bool": [true, false, true],
+                                        |    "arr_float64": [1.2, 2, 4.654],
+                                        |    "arr_empty": []
+                                        |}""".stripMargin)
+        res.isRight shouldBe true
+
+        val Right(dmiResponse) = res
+        dmiResponse.allProperties.toSet shouldBe
+          Set(
+            "externalId" -> "test",
+            "prop_float64" -> PropertyType.Float64.Property(23.0),
+            "prop_string" -> PropertyType.Text.Property("toto"),
+            "prop_direct_relation" -> PropertyType.DirectRelation.Property("Asset"),
+            "arr_bool" -> PropertyType.Array.Boolean.Property(
+              Vector(true, false, true)
+            ),
+            "arr_float64" -> PropertyType.Array.Float64.Property(
+              Vector(1.2, 2, 4.654)
+            ),
+            "arr_empty" -> PropertyType.Array.Text.Property(Vector())
+          )
+      }
+
+      "not work for primitive if given value does not match property type" in {
+        val res: Either[circe.Error, DataModelInstanceQueryResponse] =
+          decode[DataModelInstanceQueryResponse]("""{
+                                        |"modelExternalId" : "tada",
+                                        |"properties" : {
+                                        |    "prop_bool" : "true",
+                                        |    "prop_float64": 23.0,
+                                        |    "prop_string": "toto",
+                                        |    "arr_bool": [true, false],
+                                        |    "arr_float64": [1.2, 2, 4.654],
+                                        |    "arr_empty": []
+                                        |} }""".stripMargin)
+
+        checkErrorDecodingOnField(res, "prop_bool", "Boolean")
+      }
+      "not work for array if it contains Boolean and String" in {
```

Comment:
@vuhainguyen do you notice anything off with the json I provide here? I get 
```
DecodingFailure(Attempt to decode value on failed cursor, List(DownField(prop_float64), DownArray, DownField(items)))
Some((Attempt to decode value on failed cursor,List(DownField(prop_float64), DownArray, DownField(items))))
```
which is very strange as we don't have `prop_float64` in the json 🤯

Reply:
ok found the reason

Reply:
https://github.com/cognitedata/cognite-sdk-scala/pull/505/files#diff-4e0e16de6ac668b2c7f9ae85403b8f017920dd1a4c982b6971b61a4dc9eb34f0L202-L207